AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin+AuthInformation.swift

```diff
@@ -15,6 +15,9 @@ extension AWSAPIPlugin {
     }
 
     public func defaultAuthType(for apiName: String?) throws -> AWSAuthorizationType {
-        try pluginConfig.endpoints.getConfig(for: apiName).authorizationType
```

Comment:
This is a public api and you are changing the error thrown which will be a breaking change.

Reply:
I'm debating removing this change from this PR all together as it does have some behavior changes to APIPlugin when we removed the force cast to `pluginConfig` and various other properties in APIPlugin. A few details regarding this change, take `pluginConfig` for example, it was originally force cast and set when `Amplify.configure()` is called. If the developer does not call Amplify.configure and calls into one of the plugin methods like `defaultAuthType` it will crash the app. By turning it into optional, the app gracefully fails with the error indiciating that the plugin configuration not yet configured. This may happen in two cases, at development time when the developer does not call Amplify.configure or during testing when Amplify.reset is called to reset the plugins. At development time, the behavior changes from app crashing to an error thrown- which in this case is fairly similar DX, a non-recoverable error which is needs to be fixed. During testing, Amplify.reset() is called between test cases (like at the beginning of setUp or in tearDown) and i've observed that some operations are in-progress will attempt to use pluginConfig, and crash the host app. Since this PR is fairly large already, it will be easier to reason about these changes in separate PRs. This PR to only contain more granular testing done, and another PR for changes made to APIPlugin

Reply:
Removed the APIPlugin changes from this PR